storage/modules/block_storage.go

```diff
@@ -1084,6 +1136,96 @@ func (b *BlockStorage) FindTransaction(
 	return newestBlock, newestTransaction, nil
 }
 
+func (b *BlockStorage) FindRelatedTransactions(
+	ctx context.Context,
+	transactionIdentifier *types.TransactionIdentifier,
+	db database.Transaction,
+) (*types.BlockIdentifier, *types.Transaction, []*types.TransactionIdentifier, error) {
+	maxBlock, tx, err := b.FindTransaction(ctx, transactionIdentifier, db)
```

Comment:
`maxBlock` should be the value of `GetHeadBlockIdentifierTransactional` ... `FindTransaction` just returns the block that the tx was in

Reply:
which is what we want right? we want to find the latest block we need confirmed to confirm the root transaction. So we find the root block and take the highest index block of all its children.

Reply:
ahhh oops I read that check wrong 😳 ! nvm